tests/registry/registry.go

```diff
@@ -0,0 +1,39 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package registry
+
+import (
+	"github.com/onsi/ginkgo/v2"
+
+	"github.com/ava-labs/hypersdk/tests/workload"
+)
+
+type TestFunc func(t ginkgo.FullGinkgoTInterface, tn workload.TestNetwork) error
+
+type namedTest struct {
+	Fnc  TestFunc
+	Name string
+}
+type Registry struct {
+	tests []namedTest
+}
+
+func (r *Registry) Add(name string, f TestFunc) {
+	r.tests = append(r.tests, namedTest{Fnc: f, Name: name})
+}
+
+func (r *Registry) List() []namedTest {
+	return r.tests
+}
+
+var testRegistry Registry
```

Comment:
Rather than create this global here, could we leave it to each VM to create their own instance and register tests to it?

Reply:
done. wasn't as intuitive as I was hoping for, but the end results looks nice.